### PR TITLE
Enables XLSProcessor to import specific sheets from an Excel file

### PR DESCRIPTION
**Pull request by cschierle**
and changes visibility of constructor and fluent api to `public`.

Fixes: SIRI-499

### CODE REVIEW

File: src/main/java/sirius/web/data/LineBasedProcessor.java [hunk 1]
```diff
@@ -40,7 +40,10 @@ public static LineBasedProcessor create(String name, InputStream input, boolean
         if (name.toLowerCase().endsWith("csv")) {
             return new CSVProcessor(input);
         }
-        throw Exceptions.createHandled().withSystemErrorMessage("Cannot process files of type: %s", name).handle();
+        throw Exceptions.createHandled()
+                        .withNLSKey("LineBasedProcessor.error.invalidFileType")
+                        .set("type", name.substring(name.lastIndexOf(".")))
```
Comment (andyHa) [inline src/main/java/sirius/web/data/LineBasedProcessor.java:45]: Files.getFileExtension or s.th. the like....

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/web/data/XLSProcessor.java [hunk 2]
```diff
@@ -63,6 +93,7 @@ protected Workbook openWorkbook() throws IOException {
     }
 
     protected void importSheet(RowProcessor rowProcessor, Predicate<Exception> errorHandler, Sheet sheet) {
+        currentSheet = sheet.getSheetName();
```
Comment (andyHa) [inline src/main/java/sirius/web/data/XLSProcessor.java:96]: I'd not make this class stateful. Couldn't we rather subclass RowProcessor into "SheetBasedRowProcessor" which accepts an additional parameter - we could then check which class is present below and invoke the proper function...

File: src/main/java/sirius/web/data/XLSProcessor.java [hunk 3]
```diff
@@ -122,11 +153,11 @@ private Object extractCellValue(Cell cell) {
         if (cellType == CellType.BLANK) {
             return null;
         }
-        throw new IllegalArgumentException(Strings.apply(
-                "Cannot read a value of type %s from cell at row %d, column  %d",
-                cellType,
-                cell.getRowIndex(),
-                cell.getColumnIndex()));
+        throw new IllegalArgumentException(NLS.fmtr("XLSProcessor.error.invalidValueInCell")
```
Comment (andyHa) [inline src/main/java/sirius/web/data/XLSProcessor.java:156]: add sheet?

File: src/main/java/sirius/web/data/XLSProcessor.java [hunk 4]
```diff
@@ -153,4 +184,19 @@ private Object extractNumericValue(Cell cell) {
...
         }
     }
+
+    /**
+     * Specifies a list of names for sheets that are to be processed.
+     *
+     * @param sheetNames a list of sheet names to process
+     * @return the processor itself for fluent method calls
+     */
+    public XLSProcessor withSheetNames(List<String> sheetNames) {
```
Comment (andyHa) [inline src/main/java/sirius/web/data/XLSProcessor.java:194]: do we need this, if we pass the sheet along in the sheet to the RowProcessor?

File: src/main/java/sirius/web/data/XLSProcessor.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/web/data/XLSProcessor.java:96]: Or we could invoke a beginSheet(Sheet) / endSheert(Sheet) if present....

**Review by andyHa — CHANGES_REQUESTED** [review]

**Review by idlira — APPROVED** [review]
Please request review again when you work on Andi's comments. Have no opinion if your way or moving the list of sheets out of the XLSProcessor. 🤷‍♂️

File: src/main/java/sirius/web/data/XLSProcessor.java (hunk 4 above)
Comment (cschierle) [reply src/main/java/sirius/web/data/XLSProcessor.java]: no, we don't

File: src/main/java/sirius/web/data/XLSProcessor.java (hunk 2 above)
Comment (cschierle) [reply src/main/java/sirius/web/data/XLSProcessor.java]: The SheetBasedRowProcessor approach makes this a lot cleaner. Thanks for the suggestion 👍

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/web/data/XLSProcessor.java [hunk 5]
```diff
@@ -58,6 +66,42 @@ public void run(RowProcessor rowProcessor, Predicate<Exception> errorHandler) th
...
+     * @param errorHandler   errorHandler which gets called on exceptions. returns <tt>true</tt> if the exception was
+     *                       handled and <tt>false</tt> if the exception should be rethrown.
+     * @param sheetNames     the sheet names to process
+     * @throws Exception                             in case an error occurred while processing.
+     * @throws IOException                           if the stream providing the given workbook cannot be read.
+     * @throws sirius.kernel.health.HandledException if workbook does not contain a sheet with the given name.
+     */
+    public void runForSheets(SheetBasedRowProcessor sheetProcessor,
+                             Predicate<Exception> errorHandler,
+                             String... sheetNames) throws Exception {
```
Comment (andyHa) [inline src/main/java/sirius/web/data/XLSProcessor.java:82]: I'd change this to runForSheets(SheetBasedRowProcessor... processors)

And make SheetProcessor a simple class which contains a sheetName, a RowProcessor and an error Handler Predicate<Exception>

**Pull request merged**